pkg/operator/diff/internal/diff.go

```diff
@@ -186,14 +189,58 @@ func (g *DiffGenerator) Run(oldModel, newModel model.Model) (model.Model, error)
 		var outputHasDefault bool
 		outputPkg.DefaultChannel, outputHasDefault = outputPkg.Channels[newPkg.DefaultChannel.Name]
 		if !outputHasDefault {
-			// Create a name-only channel since oldModel contains the channel already.
-			outputPkg.DefaultChannel = copyChannelNoBundles(newPkg.DefaultChannel, outputPkg)
+			// Set the defaultChannel using the priority of a channel when the default got filtered out
+			// If no channels with the Priority property, raise an error
+			if err := setDefaultChannel(outputPkg); err != nil {
+				return outputModel, err
+			}
 		}
 	}
 
 	return outputModel, nil
 }
 
+type ChannelPriorityPropList []property.Channel
+
+// setDefaultChannel sets the new default channel of a package if the old default channel got filtered out.
+// Throws an error if there are no channels with the Priority property
+func setDefaultChannel(outputPkg *model.Package) error {
```

Comment:
I think this logic could use a unit test or be integrated into the existing unit tests in diff_test.go. It would help catch the errors in the cases @jchunkins is referring to.